add allure report

dbakulina opened the pull request:
Sprint_7

dbakulina committed:
add allure report

pavel-ulan commented on src/main/java/courier/CourierClient.java:
```diff
@@ -0,0 +1,148 @@
+
+                .then().log().all();
+
+    }
```
Для чего тебе пустые строки между строками кода? Есть различные компоновки, но такой точно ни в одном кодГайде нету.

pavel-ulan commented on src/main/java/courier/CourierClient.java:
```diff
@@ -0,0 +1,148 @@
+
+                .when()
+
+                .post("/courieru")
```
ну вот смотри, у тебя 2 метода рядом с запросом к /courier. Причем тут у тебя ошибка а в следующем методе нет. Как бы ты искала эту ошибку?
Плюс часть ендпоинтов у тебя вынесены в константы, часть нет. выносить - правильно и если выносить, то уже все, а не частями.

pavel-ulan commented on src/test/java/CourierTest.java:
```diff
@@ -0,0 +1,151 @@
+        courierId = courierClient.login(creds)
+                .extract().path("id");
+        assertNotEquals(0, courierId);
+        courierClient.delete(courierId);
```
Если тест не пройдет, то шаг с удалением не выполнится. С точки зрения проекта, это критическая ошибка. Подумай как это можно исправить и внеси правки.

pavel-ulan commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,138 @@
+    courierClient.create(courier);
+    CourierCredentials credsNoLogin = CourierCredentials.getWithoutLogin(courier);
+    CourierCredentials creds = CourierCredentials.from(courier);
+    courierClient.loginWrong(credsNoLogin);
```
У тебя метод loginWrong не содержит проверки. В итоге непонятно, что ты в этом тесте проверяешь.
Касательно остальных тестов, так же часто непонятно, что же проверяется в тесте. Где-то проверка зашита в метод который вызывается, где-то проверяется код ответа без использования assert (да, так можно, но выглядит это странно. тем более, что rest-assured предоставляет точку доступа к assertions  через свои методы).

pavel-ulan commented on src/test/java/OrderTest.java:
```diff
@@ -0,0 +1,142 @@
+ // 1.Проверить что можно указать один из цветов — BLACK ;
+    @Test
+    public void createOrderBlack(){
+        File json = new File("src/test/resources/order.json");
```
Похвально, вариант загрузить тестовые данные из файла встречается нечасто.

pavel-ulan reviewed (changes requested):
Работа неплохая. Есть ряд вопросов как к компоновке тестов, так и к самим проверкам. Посмотри комментарии и поправь. И пожалуйста, убери пустые строчки. пропуск строки допустим между методами, после полей класса, ну и иногда что бы разделить блоки кода. Но точно не так как в работе сейчас.

dbakulina committed:
Замечание 1: удалила пустые строки

dbakulina committed:
Замечание 2: вынесла все эндпоинты в константы
Метод createWrongRoot используется в тесте courierCreateWrongRootStatus, который проверяет код ответа при неверном адресе запроса.
Второй метод createWrongMethod используется в тесте courierCreateWrongMethodStatus и проверяет код ответа при неверном методе.

dbakulina committed:
Замечание 2: вынесла все эндпоинты в константы
Метод createWrongRoot используется в тесте courierCreateWrongRootStatus, который проверяет код ответа при неверном адресе запроса.
Второй метод createWrongMethod используется в тесте courierCreateWrongMethodStatus и проверяет код ответа при неверном методе.

dbakulina committed:
Замечание 3: Вынесла удаление курьера в @After, теперь курьер удалится в любом случае. Так как есть тесты, в которых курьер не создается, то я вынесла их в отдельный класс.

dbakulina committed:
Замечание 4: Везде вынесла проверки в тестах через assert

pavel-ulan reviewed (approved):
Еще совет. Почитай что стоит коммитить, а что не стоит коммитить в репозиторий (так .idea не коммитится например).
В остальном неплохо, правки вижу, в целом, молодец.
